src/runtime_src/core/tools/xbmgmt2/flash/xqspips.cpp

```diff
@@ -482,6 +482,24 @@ unsigned int XQSPIPS_Flasher::getFlashSize()
 
 void XQSPIPS_Flasher::readBack(const std::string& output, unsigned int base)
 {
+    initQSpiPS();
+
+    uint32_t StatusReg = XQSpiPS_GetStatusReg();
+
+    if (StatusReg == 0xFFFFFFFF) {
+        std::cout << "[ERROR]: Read PCIe device return -1. Cannot get QSPI status." << std::endl;
```

Comment:
Anytime xbmgmt produces an error, it should return a non-zero error value.

This function should be updated to reflect that.   If there isn't time in the release to do this, then please create a CR so that this can be done in the next release.

Reply:
agree.
in today's code, the caller from xbmgmt is also void, and hence it makes no sense to return errno.
I can file CR tracking this. whoever changes the xbmgmt can change this API also

Reply:
https://jira.xilinx.com/browse/CR-1129024
is logged